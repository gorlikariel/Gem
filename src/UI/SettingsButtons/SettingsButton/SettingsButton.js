import React from "react";
import { Link } from "react-router-dom";
import WideButton from "../../WideButton/WideButton";
import { Typography } from "@material-ui/core";
import AccountIcon from "../../Icons/AccountIcon/AccountIcon";
import ClockIcon from "../../Icons/ClockIcon/ClockIcon";
import PillIcon from "../../Icons/PillIcon/PillIcon";
import * as theme from "../../theme/theme";
import RightArrow from "../../Icons/RightArrow/RightArrow";
const SettingsButton = props => {
  const { bgColor, color, to, children, icon } = props;
  const clicked = () => {
    console.log("FUUUCK");
  };
  const rightArrow = <RightArrow style={{ paddingRight: "12px" }} />;
  const styles = {
    buttonIcon: {
      paddingLeft: "20px"
    },
    logoutButton: {
      width: "24px",
      paddingLeft: "20px"
    },
    navLink: {
      textDecoration: "none",
      color: "inherit"
    }
  };
  const icons = {
    accountIcon: <AccountIcon style={styles.buttonIcon} />,
    alarmIcon: <ClockIcon style={styles.buttonIcon} />,
    pillIcon: <PillIcon style={styles.buttonIcon} />
  };

  return (
    <Link style={{ textDecoration: "none" }} to={to}>
      <WideButton bgColor={theme[bgColor]} color={theme[color]}>
        {icons[icon]}
        <Typography
          style={{ paddingLeft: to === "/" ? "54px" : "12px" }}
          variant="display3"
          color="inherit"
        >
          {children}
        </Typography>
        <div style={{ marginLeft: "auto" }}>
          {to === "/" ? null : rightArrow}
        </div>
      </WideButton>
    </Link>
  );
};
export default SettingsButton;
