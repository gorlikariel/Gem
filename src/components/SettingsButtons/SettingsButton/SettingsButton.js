import React from 'react';
import { Link } from 'react-router-dom';
import WideButton from '../../WideButton/WideButton';
import { Typography, Collapse } from '@material-ui/core';
import * as theme from '../../../styleguide/theme';
import AccountIcon from '../../../Icons/AccountIcon/AccountIcon';
import RightArrow from '../../../Icons/RightArrow/RightArrow';
import PillIcon from '../../../Icons/PillIcon/PillIcon';
import ClockIcon from '../../../Icons/ClockIcon/ClockIcon';
const SettingsButton = props => {
  const { bgColor, color, to, children, icon } = props;
  const rightArrow = <RightArrow style={{ paddingRight: '12px' }} />;
  const styles = {
    buttonIcon: {
      paddingLeft: '20px'
    },
    logoutButton: {
      width: '24px',
      paddingLeft: '20px'
    }
  };
  const icons = {
    accountIcon: <AccountIcon style={styles.buttonIcon} />,
    alarmIcon: <ClockIcon style={styles.buttonIcon} />,
    pillIcon: <PillIcon style={styles.buttonIcon} />
  };

  return (
    <Collapse in={props.in} timeout={props.timeout}>
      <Link style={{ textDecoration: 'none' }} to={to}>
        <div
          style={{
            borderImage: ' linear-gradient(to left, #743ad5 0%, #d53a9d 100%)',
            borderImageSlice: 1,
            borderWidth: '10px'
          }}
        >
          <WideButton
            onClick={props.onClick}
            bgColor={bgColor}
            color={color}
            onClick={props.onClick}
            noBoxShadow
          >
            {icons[icon]}
            <Typography
              style={{ paddingLeft: to === '/' ? '55px' : '12px' }}
              variant="display3"
              color="inherit"
            >
              {children}
            </Typography>
            <div style={{ marginLeft: 'auto' }}>
              {to === '/' ? null : rightArrow}
            </div>
          </WideButton>
        </div>
      </Link>
    </Collapse>
  );
};
export default SettingsButton;
