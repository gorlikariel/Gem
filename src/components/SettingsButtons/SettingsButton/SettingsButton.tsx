import * as React from 'react';
import { Link } from 'react-router-dom';
import WideButton from '../../WideButton/WideButton';
import { Typography, Collapse } from '@material-ui/core';
import AccountIcon from '../../../Icons/AccountIcon/AccountIcon';
import RightArrow from '../../../Icons/RightArrow/RightArrow';
import PillIcon from '../../../Icons/PillIcon/PillIcon';
import ClockIcon from '../../../Icons/ClockIcon/ClockIcon';
interface SettingsButtonProps {
  bgColor: string;
  color: string;
  to: string;
  icon?: string;
  timeout?: number;
  onClick?: React.EventHandler<any>;
  in?: boolean;
}

const SettingsButton: React.SFC<SettingsButtonProps> = props => {
  const { bgColor, color, to, children, icon, timeout, onClick } = props;
  const rightArrow = <RightArrow style={{ paddingRight: '12px' }} />;
  const styles = {
    buttonIcon: {
      paddingLeft: '20px'
    },
    logoutButton: {
      width: '24px',
      paddingLeft: '20px'
    },
    rightArrow: { marginLeft: 'auto' },
    buttonText: { paddingLeft: to === '/' ? '55px' : '12px' },
    link: { textDecoration: 'none' }
  };
  const icons = {
    accountIcon: <AccountIcon style={styles.buttonIcon} />,
    alarmIcon: <ClockIcon style={styles.buttonIcon} />,
    pillIcon: <PillIcon style={styles.buttonIcon} />
  };

  return (
    <Collapse in={props.in} timeout={timeout}>
      <Link style={styles.link} to={to}>
        <WideButton onClick={onClick} bgColor={bgColor} color={color}>
          {icon ? icons[icon] : null}
          <Typography
            style={styles.buttonText}
            variant="display3"
            color="inherit"
          >
            {children}
          </Typography>
          <div style={styles.rightArrow}>{to === '/' ? null : rightArrow}</div>
        </WideButton>
      </Link>
    </Collapse>
  );
};
export default SettingsButton;
