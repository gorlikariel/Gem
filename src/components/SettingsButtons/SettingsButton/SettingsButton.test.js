import React from 'react';
import { shallow, EnzymeAdapter, configure } from 'enzyme';
import SettingsButton from './SettingsButton';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
test('settings button should render text as expected', () => {
  const settingsButton = shallow(
    <SettingsButton to="someWhere" bgImg="red" color="red">
      Test Text
    </SettingsButton>
  );
  const tree = toJson(settingsButton);
  expect(tree).toMatchSnapshot();
});
