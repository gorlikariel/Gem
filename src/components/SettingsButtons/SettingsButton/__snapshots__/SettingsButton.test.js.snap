// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings button should render text as expected 1`] = `
<WithStyles(Collapse)>
  <div
    style={
      Object {
        "paddingBottom": "10px",
        "paddingLeft": "5px",
        "paddingRight": "5px",
        "paddingTop": "10px",
      }
    }
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="someWhere"
    >
      <WideButton
        bgImg="red"
        color="red"
        height="64px"
        marginTop="0px"
        noBoxShadow={false}
        type="button"
        width="100%"
      >
        <WithStyles(Typography)
          color="inherit"
          style={
            Object {
              "paddingLeft": "12px",
            }
          }
          variant="display3"
        >
          Test Text
        </WithStyles(Typography)>
        <div
          style={
            Object {
              "marginLeft": "auto",
            }
          }
        >
          <WithStyles(LeftArrow)
            style={
              Object {
                "paddingRight": "12px",
              }
            }
          />
        </div>
      </WideButton>
    </Link>
  </div>
</WithStyles(Collapse)>
`;
